### PR TITLE
Fixes #286 bug that occured on Windows

### LINKED ISSUE #286: [Windows] Error in plugin 'gulp-less' when executing 'npm run gulp'
### Bug report

- __Package version(s)__: 1.1.0
- __Browser and OS versions__: Win 7 x64

#### Steps to reproduce

Did as instructed here:
https://github.com/palantir/blueprint/blob/master/README.md#development
> 1. `git clone` this repository (or fork if you lack permissions)
> 1. `npm install` to install build dependencies
> 1. `npm run bootstrap` to install and link each package using [Lerna](https://lernajs.io/)
> 1. `npm run gulp` to compile and start the server and watcher

Last step failed and produced the error you can read about below.

I also updated Node to its current version (v7.2.0) - it didn't help at all.

#### Actual behavior

```
C:\Users\Admin\Desktop\bp\blueprint>npm run gulp

> blueprint@1.1.0 gulp C:\Users\Admin\Desktop\bp\blueprint
> gulp

[01:36:03] Using gulpfile ~\Desktop\bp\blueprint\gulpfile.js
[01:36:03] Starting 'default'...
[01:36:03] Starting 'clean'...
[01:36:03] Finished 'clean' after 41 ms
[01:36:03] Starting 'compile'...
[01:36:03] Starting 'icons'...
[01:36:03] Finished 'icons' after 85 ms
[01:36:03] Starting 'sass-variables'...
[01:36:04] 'sass-variables' errored after 689 ms
[01:36:04] Error in plugin 'gulp-less'
Message:
    'mixins.less' wasn't found. Tried - C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\mixins.less,mixins.less in file C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less line no. 185
Details:
    type: File
    filename: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
    index: 4082
    line: 185
    callLine: NaN
    callExtract: undefined
    column: 0
    extract: @import "colors";,@import "mixins";,
    lineNumber: 185
    fileName: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
[01:36:04] 'default' errored after 831 ms
[01:36:04] Error in plugin 'run-sequence(sass-variables)'
Message:
    'mixins.less' wasn't found. Tried - C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\mixins.less,mixins.less in file C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less line no. 185
Details:
    type: File
    filename: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
    index: 4082
    line: 185
    callLine: NaN
    callExtract: undefined
    column: 0
    extract: @import "colors";,@import "mixins";,
    lineNumber: 185
    fileName: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
Stack:
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:56:13)
  at Gulp.onError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:67:4)
  at emitOne (events.js:101:20)
  at Gulp.emit (events.js:188:7)
  at Gulp.Orchestrator._emitTaskDone (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:264:8)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:275:23
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:21:8)
  at DestroyableTransform.<anonymous> (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:52:4)
  at DestroyableTransform.f (C:\Users\Admin\Desktop\bp\blueprint\node_modules\end-of-stream\node_modules\once\once.js:17:25)
  at emitOne (events.js:101:20)
  at DestroyableTransform.emit (events.js:188:7)
  at onwriteError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:356:10)
  at onwrite (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:373:11)
  at WritableState.onwrite (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:126:5)
  at afterTransform (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_transform.js:81:3)
  at TransformState.afterTransform (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_transform.js:58:12)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\gulp-less\index.js:58:14
  at tryCatchReject (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:845:30)
  at runContinuation1 (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:804:4)
  at Rejected.when (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:625:4)
  at Pending.run (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:483:13)
  at Scheduler._drain (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\Scheduler.js:62:19)
  at Scheduler.drain (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\Scheduler.js:27:9)
  at _combinedTickCallback (internal/process/next_tick.js:67:7)
  at process._tickCallback (internal/process/next_tick.js:98:9)

[01:36:04] 'compile' errored after 788 ms
[01:36:04] Error in plugin 'run-sequence(default)'
Message:
    'mixins.less' wasn't found. Tried - C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\mixins.less,mixins.less in file C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less line no. 185
Details:
    type: File
    filename: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
    index: 4082
    line: 185
    callLine: NaN
    callExtract: undefined
    column: 0
    extract: @import "colors";,@import "mixins";,
    lineNumber: 185
    fileName: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
Stack:
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:56:13)
  at Gulp.onError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:67:4)
  at emitOne (events.js:101:20)
  at Gulp.emit (events.js:188:7)
  at Gulp.Orchestrator._emitTaskDone (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:264:8)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:275:23
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:21:8)
  at cb (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:29:3)
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:60:5)
  at Gulp.onError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:67:4)
  at emitOne (events.js:101:20)
  at Gulp.emit (events.js:188:7)
  at Gulp.Orchestrator._emitTaskDone (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:264:8)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:275:23
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:21:8)
  at DestroyableTransform.<anonymous> (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:52:4)
  at DestroyableTransform.f (C:\Users\Admin\Desktop\bp\blueprint\node_modules\end-of-stream\node_modules\once\once.js:17:25)
  at emitOne (events.js:101:20)
  at DestroyableTransform.emit (events.js:188:7)
  at onwriteError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:356:10)
  at onwrite (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:373:11)
  at WritableState.onwrite (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:126:5)
  at afterTransform (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_transform.js:81:3)
  at TransformState.afterTransform (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_transform.js:58:12)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\gulp-less\index.js:58:14
  at tryCatchReject (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:845:30)
  at runContinuation1 (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:804:4)
  at Rejected.when (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:625:4)
  at Pending.run (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:483:13)
  at Scheduler._drain (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\Scheduler.js:62:19)
  at Scheduler.drain (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\Scheduler.js:27:9)
  at _combinedTickCallback (internal/process/next_tick.js:67:7)
  at process._tickCallback (internal/process/next_tick.js:98:9)

[01:36:04] 'compile' errored after 797 ms
[01:36:04] Error in plugin 'run-sequence(sass-variables)'
Message:
    'mixins.less' wasn't found. Tried - C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\mixins.less,mixins.less in file C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less line no. 185
Details:
    type: File
    filename: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
    index: 4082
    line: 185
    callLine: NaN
    callExtract: undefined
    column: 0
    extract: @import "colors";,@import "mixins";,
    lineNumber: 185
    fileName: C:\Users\Admin\Desktop\bp\blueprint\packages\core\dist\variables.less
Stack:
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:56:13)
  at Gulp.onError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\run-sequence\index.js:67:4)
  at emitOne (events.js:101:20)
  at Gulp.emit (events.js:188:7)
  at Gulp.Orchestrator._emitTaskDone (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:264:8)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\index.js:275:23
  at finish (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:21:8)
  at DestroyableTransform.<anonymous> (C:\Users\Admin\Desktop\bp\blueprint\node_modules\orchestrator\lib\runTask.js:52:4)
  at DestroyableTransform.f (C:\Users\Admin\Desktop\bp\blueprint\node_modules\end-of-stream\node_modules\once\once.js:17:25)
  at emitOne (events.js:101:20)
  at DestroyableTransform.emit (events.js:188:7)
  at onwriteError (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:356:10)
  at onwrite (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:373:11)
  at WritableState.onwrite (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_writable.js:126:5)
  at afterTransform (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_transform.js:81:3)
  at TransformState.afterTransform (C:\Users\Admin\Desktop\bp\blueprint\node_modules\readable-stream\lib\_stream_transform.js:58:12)
  at C:\Users\Admin\Desktop\bp\blueprint\node_modules\gulp-less\index.js:58:14
  at tryCatchReject (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:845:30)
  at runContinuation1 (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:804:4)
  at Rejected.when (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:625:4)
  at Pending.run (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\makePromise.js:483:13)
  at Scheduler._drain (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\Scheduler.js:62:19)
  at Scheduler.drain (C:\Users\Admin\Desktop\bp\blueprint\node_modules\when\lib\Scheduler.js:27:9)
  at _combinedTickCallback (internal/process/next_tick.js:67:7)
  at process._tickCallback (internal/process/next_tick.js:98:9)


npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "gulp"
npm ERR! node v6.6.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! blueprint@1.1.0 gulp: `gulp`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the blueprint@1.1.0 gulp script 'gulp'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the blueprint package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs blueprint
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls blueprint
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Admin\Desktop\bp\blueprint\npm-debug.log
```

#### Expected behavior

_No idea. Didn't happen to see it working so far._

### PR DESCRIPTION
**Pull request by Nonemoticoner**
#### Fixes #286 

I was unable to test it on Mac so it might actually ruin it in exchange for fixing it on Windows.

#### Changes proposed in this pull request:

- Correct interpretation of new line on Windows in regex expression ran by Gulp

#### Reviewers should focus on:

- Compatibility with Mac

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @Nonemoticoner! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by Nonemoticoner** [comment]
Seems like tests ran by CircleCI have just confirmed that this bug is a special Windows case and there is a requirement for differentiating what (Gulp) task should be ran depending on platform.

File: gulp/sass.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ module.exports = (gulp, plugins, blueprint) => {
             // package the variables list for consumption -- no imports or functions
             .pipe(plugins.stripCssComments())
             .pipe(plugins.replace(/\n{3,}/g, "\n\n"))
-            .pipe(plugins.replace(/(@import|\/\/).*\n+/g, ""))
+            .pipe(plugins.replace(/(@import|\/\/).*\r\n+/g, ""))
```
Comment (giladgray) [inline gulp/sass.js]: this change doesn't work on Mac now! can you try change it to `[\r\n]+`? that says "at least one `\r` _or_ `\n`"

Comment (cmslewis) [inline gulp/sass.js]: What if we use `\r?\n` to indicate that the `\r` is optional but the `\n` is required?

Comment (giladgray) [inline gulp/sass.js]: maybe we can actually remove the `\n` from this expression entirely and instead capture it in the `\n{3,}` rule? would be better to handle newlines in fewer places.

**Comment by giladgray** [comment]
Closing as there's been no recent activity and Windows development is extraordinarily low priority.

This should be relatively easy to polish (just need to enhance the regex to support both UNIX and Windows line endings) but the team doesn't have the bandwidth or urgency.

**Pull request closed without merging**